harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -359,6 +359,15 @@ def remove_metadata(self, keys: List[str]) -> None:
 
         self._push_metadata()
 
+    def delete(self) -> None:
+        """
+        Deletes checkpoint and all associated data in the checkpoint storage.
```

Comment:
prolly explain that it will tell the master and the master will run it asynchronously